.travis.yml

```diff
@@ -18,8 +18,10 @@ before_install:
 - echo HRNGDEVICE=/dev/urandom | sudo tee /etc/default/rng-tools
 - sudo /etc/init.d/rng-tools restart
 install:
-- pip install ansible==1.8.4
-- pip install coveralls
+- pip freeze -l
+# Preinstalled travis pytest is not tracked by us
+- pip uninstall pytest -y
+- pip install -r testinfra/requirements.txt
```

Comment:
I think shoehorning is the wrong word here when these both these requirements files are for test dependencies. App code and spec tests are both important and should be run not only on our development machines, but staging and prod as well. The requirements need to be installed for that to happen.

The more requirements lists we create, the greater room there is for conflict. testinfra should catch these, but we still have to resolve them manually. That's one of the reasons we use `pip-tools`, but even that approach is complicated by too many requirements lists. (Speaking of which, either way you should be using `pip-tools` in order to lock down your second-level dependencies for reproducible tests).

I think that isolation of the testing and app code dependencies is a good thing, that may be well solved by your suggestion of running tests from a separate machine. I think this is something we need to tackle separately, however. Adding `pytest-xdist` and `testinfra` to our `test-requirements.in` (everything else in `testinfra/requirements.txt` is already there except Ansible, which could be installed on its own) does not seem like a problematic move when afaik this lack of isolation has only been a problem in theory.

On the other hand, it would (i) be a sensible logical grouping, (ii) enable us to use `pip-tools` and our existing `pip_update.sh` script to update these periodically and easily along with our other requirements, and (iii) make it easier for developers to just run the testinfra suite on the staging and prod machines--if the tools aren't just there, we'll be less likely to go to the annoying process of installing them separately, and the testinfra suite will get less love and less use on these machines.